inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.identity.authenticator;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+
+public class CheckPushResult {
+    private static final Log log = LogFactory.getLog(CheckPushResult.class);
+
+    private static String urlString = InweboConstants.INWEBOURL;
+    private String serviceId;
+    private String p12file;
+    private String p12password;
+    private SSLContext context = null;
+
+    public CheckPushResult(String id, String p12file, String p12password) {
+        this.serviceId = id;
+        this.p12file = p12file;
+        this.p12password = p12password;
+    }
+
+    /**
+     * Prompt for a login and an OTP and check if they are OK.
+     */
+    public JSONObject checkPushResult(String userId, String sessionId) {
+        String urlParameters = null;
+        JSONObject json = null;
+        try {
```

Comment:
don't use several try catch statements here. Try to use try catch statements in more readable manner